crates/net/downloaders/src/receipt_file_client.rs

```diff
@@ -65,7 +66,16 @@ impl FromReader for ReceiptFileClient {
                             bytes_len=bytes.len(),
                             "partial receipt returned from decoding chunk"
                         );
-                        remaining_bytes = bytes;
+
+                        // Empty list code indicates no receipts for the block, and no other receipt
+                        // info
+                        if !bytes.is_empty() && bytes[0] == EMPTY_LIST_CODE {
+                            // empty list, so no receipts, just advance
+                            remaining_bytes = bytes[1..].into();
+                        } else {
+                            remaining_bytes = bytes;
+                        }
```

Comment:
will move this to the codec. not only because it makes more sense there, but also because it may be smthg specific to testinprod's `HackReceipt` encoding. moving it allows `ReceiptFileClient` to easily be made generic over the codec in the future, if that will be needed.